devicetypes/smartthings/orvibo-smoke-sensor.src/orvibo-smoke-sensor.groovy

```diff
@@ -0,0 +1,161 @@
+	}
+}
+
+def installed(){
+	log.debug "installed"
+	refresh()
+}
+
+def parse(String description) {
+	log.debug "description(): $description"
+	def map = zigbee.getEvent(description)
+	if(!map){
+		if (description?.startsWith('zone status')) {
+			map = parseIasMessage(description)
+		}else{
+			map = parseAttrMessage(description)
+		}
+	}
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+	if (description?.startsWith('enroll request')) {
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+		result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+	}
+	return result
+}
+
+def parseAttrMessage(String description){
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	def map = [:]
+	if (descMap?.clusterInt == 0x0009 && descMap.value) {
+		map = getAlarmResult(descMap.value == "0000" ? false :true)
+	}else if(descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+		map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+	}else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+		def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+		map = translateZoneStatus(zs)
+	}
+	return map;
+}
+
+def parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	return getDetectedResult(zs.isAlarm1Set() || zs.isAlarm2Set())
+}
+
+private Map translateZoneStatus(ZoneStatus zs) {
+	return getDetectedResult(zs.isAlarm1Set() || zs.isAlarm2Set())
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+	log.debug "Battery Percentage rawValue = ${rawValue} -> ${rawValue / 2}%"
+	def result = [:]
+
+	if (0 <= rawValue && rawValue <= 200) {
+		result.name = 'battery'
+		result.translatable = true
+		result.value = Math.round(rawValue / 2)
+		result.descriptionText = "${device.displayName} battery was ${result.value}%"
+	}
+
+	return result
+}
+def getDetectedResult(value) {
+	def detected = value ? 'detected': 'clear'
+	String descriptionText = "${device.displayName} smoke ${detected}"
+	return [name:'smoke',
+			value: detected,
+			descriptionText:descriptionText,
+			translatable:true]
+}
+def getAlarmResult(value) {
+	def alarm = value ? 'siren': 'off'
+	String descriptionText = "${device.displayName} alarm  ${alarm}"
+	return [name:'alarm',
+			value: alarm,
+			descriptionText:descriptionText,
+			translatable:true]
+}
+def siren() {
+	sendEvent(name: "alarm", value: "siren")
+}
+def off() {
+	sendEvent(name: "alarm", value: "off")
+}
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021) +
+					zigbee.readAttribute(0x0009,0x0000) +
+					zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER,zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+	return refreshCmds
+}
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping "
+	refresh()
```

Comment:
For IAS Zone devices we typically only read `zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS)` in ping.